external/libopus/Makefile.in

```diff
@@ -0,0 +1,3386 @@
+psdir = @psdir@
+sbindir = @sbindir@
+sharedstatedir = @sharedstatedir@
+srcdir = @srcdir@
+sysconfdir = @sysconfdir@
+target_alias = @target_alias@
+top_build_prefix = @top_build_prefix@
+top_builddir = @top_builddir@
+top_srcdir = @top_srcdir@
+AUTOMAKE_OPTIONS = subdir-objects
+ACLOCAL_AMFLAGS = -I m4
+lib_LTLIBRARIES = libopus.la
+DIST_SUBDIRS = doc
+AM_CPPFLAGS = -I$(top_srcdir)/include -I$(top_srcdir)/celt -I$(top_srcdir)/silk \
+              -I$(top_srcdir)/silk/float -I$(top_srcdir)/silk/fixed $(NE10_CFLAGS)
+
+CELT_SOURCES = celt/bands.c celt/celt.c celt/celt_encoder.c \
+	celt/celt_decoder.c celt/cwrs.c celt/entcode.c celt/entdec.c \
+	celt/entenc.c celt/kiss_fft.c celt/laplace.c celt/mathops.c \
+	celt/mdct.c celt/modes.c celt/pitch.c celt/celt_lpc.c \
+	celt/quant_bands.c celt/rate.c celt/vq.c $(am__append_7) \
+	$(am__append_8) $(am__append_9) $(am__append_10) \
+	$(am__append_12) $(am__append_14)
+CELT_SOURCES_SSE = \
+celt/x86/x86cpu.c \
+celt/x86/x86_celt_map.c \
+celt/x86/pitch_sse.c
+
+CELT_SOURCES_SSE2 = \
+celt/x86/pitch_sse2.c \
+celt/x86/vq_sse2.c
+
+CELT_SOURCES_SSE4_1 = \
+celt/x86/celt_lpc_sse.c \
+celt/x86/pitch_sse4_1.c
+
+CELT_SOURCES_ARM = \
+celt/arm/armcpu.c \
+celt/arm/arm_celt_map.c
+
+CELT_SOURCES_ARM_ASM = \
+celt/arm/celt_pitch_xcorr_arm.s
+
+CELT_AM_SOURCES_ARM_ASM = \
+celt/arm/armopts.s.in
+
+CELT_SOURCES_ARM_NEON_INTR = \
+celt/arm/celt_neon_intr.c \
+celt/arm/pitch_neon_intr.c
+
+CELT_SOURCES_ARM_NE10 = \
+celt/arm/celt_ne10_fft.c \
+celt/arm/celt_ne10_mdct.c
+
+SILK_SOURCES = silk/CNG.c silk/code_signs.c silk/init_decoder.c \
+	silk/decode_core.c silk/decode_frame.c \
+	silk/decode_parameters.c silk/decode_indices.c \
+	silk/decode_pulses.c silk/decoder_set_fs.c silk/dec_API.c \
+	silk/enc_API.c silk/encode_indices.c silk/encode_pulses.c \
+	silk/gain_quant.c silk/interpolate.c silk/LP_variable_cutoff.c \
+	silk/NLSF_decode.c silk/NSQ.c silk/NSQ_del_dec.c silk/PLC.c \
+	silk/shell_coder.c silk/tables_gain.c silk/tables_LTP.c \
+	silk/tables_NLSF_CB_NB_MB.c silk/tables_NLSF_CB_WB.c \
+	silk/tables_other.c silk/tables_pitch_lag.c \
+	silk/tables_pulses_per_block.c silk/VAD.c \
+	silk/control_audio_bandwidth.c silk/quant_LTP_gains.c \
+	silk/VQ_WMat_EC.c silk/HP_variable_cutoff.c silk/NLSF_encode.c \
+	silk/NLSF_VQ.c silk/NLSF_unpack.c silk/NLSF_del_dec_quant.c \
+	silk/process_NLSFs.c silk/stereo_LR_to_MS.c \
+	silk/stereo_MS_to_LR.c silk/check_control_input.c \
+	silk/control_SNR.c silk/init_encoder.c silk/control_codec.c \
+	silk/A2NLSF.c silk/ana_filt_bank_1.c silk/biquad_alt.c \
+	silk/bwexpander_32.c silk/bwexpander.c silk/debug.c \
+	silk/decode_pitch.c silk/inner_prod_aligned.c silk/lin2log.c \
+	silk/log2lin.c silk/LPC_analysis_filter.c \
+	silk/LPC_inv_pred_gain.c silk/table_LSF_cos.c silk/NLSF2A.c \
+	silk/NLSF_stabilize.c silk/NLSF_VQ_weights_laroia.c \
+	silk/pitch_est_tables.c silk/resampler.c \
+	silk/resampler_down2_3.c silk/resampler_down2.c \
```

Comment:
hmm it contains its own resampler??

Reply:
Yes, opus has it's own resampler, so that when decoding user can specify the desired channel-num and sample-rate, and opus outputs the desired PCM data.

Reply:
Then how we can select resampler between our resampler and opus's ?